calico/getting-started/kubernetes/vpp/specificities.md

```diff
@@ -0,0 +1,25 @@
+---
+title: VPP implementation specificities & known-issues
+description: Behavioral discrepencies when running with the Calico/VPP dataplane
+canonical_url: '/getting-started/kubernetes/vpp/specificities'
+---
+
+Enabling VPP as the calico Dataplane should be transparent for most applications, but some specific behaviours might differ. This page gives a summary of the main differences, as well as the features that are still unsupported or with known issues.
+
+### Behavioural specificities
+
+The main difference between VPP and a regular iptables/IPVS dataplane is in the NodePorts implementation. As the constraints differ, it allows VPP to optimise the service implementation, but as a consequence, some behaviours differ. This will mostly impact policies expecting packets to have been source NATed or not.
+
+* For ``ClusterIPs``, ``ExternalIPs`` and ``LoadBalancerIPs`` load-balancing is done with the Maglev algorithm, and the packets are only NAT-ed on the node where the selected backend lives. This allows us to avoid source NAT-ing packets, and thus presenting the real client address to the destination pod. The same is true when a pod connects to a ClusterIP. This behavior allows the service load balancing to use direct service return (DSR) by default.
```

Comment:
```suggestion
* For ``ClusterIPs``, ``ExternalIPs`` and ``LoadBalancerIPs`` load-balancing is done with the Maglev algorithm, and the packets are only NAT-ed on the node where the selected backend lives. This allows us to avoid source NAT-ing packets, and thus present the real client address to the destination pod. The same is true when a pod connects to a ClusterIP. This behavior allows the service load balancing to use direct service return (DSR) by default.
```